core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -107,6 +104,11 @@
     protected String containerId;
     protected String containerName;
 
+    /**
+     * The approach to determine if the container is ready.
+     */
+    protected WaitStrategy waitStrategy;
```

Comment:
Set to `Wait.defaultWaitStrategy()` and mark @ NonNull on the field and `withWaitStrategy` parameter? Then no need for a null check at time of use...